### PR TITLE
ARROW-1278: [Integration] Adding integration tests for fixed_size_list

### PR DESCRIPTION
**Pull request by bkietz**
Adds integration tests for fixed_size_list
Also adds support for fixed_size_list to RecordBatchSerializer, which was omitted in #4278

### CODE REVIEW

**Comment by bkietz** [comment]
waiting for https://github.com/apache/arrow/pull/4316 to be merged

**Comment by wesm** [comment]
Is it expected for the integration tests to pass (there is support in Java for reading these?)?

**Comment by bkietz** [comment]
It seems like there is support: https://github.com/apache/arrow/blob/master/java/vector/src/test/java/org/apache/arrow/vector/ipc/TestArrowFile.java#L623

I'd need to spend more time figuring out exactly which java code is exercised by `integration_test.py`

**Comment by wesm** [comment]
I just poked around the Java codebase. It seems it may work by accident, but it is not tested

**Comment by emkornfield** [comment]
@bkietz what are the next steps here, do you need help with something on the Java side?

**Comment by bkietz** [comment]
@emkornfield I have investigated the fixed size list tests. They're spare but present so I think we should merge this and add a JIRA to test fixed size list more thoroughly in Java.

**Comment by emkornfield** [comment]
@bkietz sounds good, can you rebase?

**Comment by bkietz** [comment]
@emkornfield rebased and green except for a [coredump in test_csv](https://travis-ci.org/apache/arrow/jobs/544810176#L2195) (I don't *think* that was me)

File: cpp/Dockerfile [hunk 1]
```diff
@@ -26,7 +26,6 @@ RUN export DEBIAN_FRONTEND=noninteractive && \
         g++ \
         gcc \
         git \
-        ninja-build \
```
Comment (emkornfield) [inline cpp/Dockerfile]: why this change?

**Comment by emkornfield** [comment]
@bkietz one small question about a change to docker file.  Otherwise LGTM

File: cpp/Dockerfile (hunk 1 above)
Comment (bkietz) [reply cpp/Dockerfile]: I think that's a holdover from trying to run integration tests in docker; at one point I got `ninja: command not found`. It's not relevant here so I'll remove it

Comment (bkietz) [reply cpp/Dockerfile]: @emkornfield ninja is installed [in `ci/conda_env_cpp.yml`](https://github.com/apache/arrow/blob/a4dad3277f32b79e7da2488846f68e1987400786/ci/conda_env_cpp.yml#L33) so isn't this unused?

**Comment by emkornfield** [comment]
@bkietz potentially not an expert on docker stuff, so I'd prefer cleanup as a separate PR

**Comment by bkietz** [comment]
@emkornfield reverted and green

**Comment by emkornfield** [comment]
+1 LGTM, please file a JIRA for the follow-up work.
